src/platform/checksum.cpp

```diff
@@ -0,0 +1,43 @@
+#include "checksum.h"
+#include "package.h"
+#include "i_platform.h"
+#include <boost/algorithm/string.hpp>
+#include <fstream>
+#include <sstream>
+
+namespace platform {
+    // calculates checksum on the given file, to make it multiplatform EOL characters are ripped before checksum
+    boost::uint32_t FileChecksum( std::string const& filename )
```

Comment:
I'd just add this to ifile.*; using readall, removeeol, and checksumming. Then it works for osfiles and packaged files as well.